File: mobatech-flutter/lib/features/appointment/data/models/appointment.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'doctor.dart';
import 'doctor_schedule.dart';

part 'appointment.freezed.dart';
part 'appointment.g.dart';

@freezed
abstract class Appointment with _$Appointment {
  const factory Appointment({
    @JsonKey(name: 'ID') @Default(0) int id,
    @JsonKey(name: 'user_id') @Default(0) int userId,
    @JsonKey(name: 'doctor_id') @Default(0) int doctorId,
    @JsonKey(name: 'doctor_schedule_id') @Default(0) int doctorScheduleId,
    @Default('') String status,
    @Default('') String notes,
    @JsonKey(includeToJson: false) Doctor? doctor,
    @JsonKey(includeToJson: false) DoctorSchedule? schedule,
  }) = _Appointment;

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);
}

File: mobatech-backend/models/appointment.go
package models

import "gorm.io/gorm"

type Appointment struct {
	gorm.Model
	UserID           uint            `json:"user_id" gorm:"index"`
	User             *User           `json:"user,omitempty" gorm:"foreignKey:UserID"`
	DoctorID         uint            `json:"doctor_id" gorm:"index"`
	Doctor           *Doctor         `json:"doctor,omitempty" gorm:"foreignKey:DoctorID"`
	DoctorScheduleID uint            `json:"doctor_schedule_id" gorm:"index"`
	Schedule         *DoctorSchedule `json:"schedule,omitempty" gorm:"foreignKey:DoctorScheduleID"`
	Status             string          `json:"status" gorm:"default:'pending';index"` // pending, approved, cancelled, completed
	Notes              string          `json:"notes"`
	CancellationReason string          `json:"cancellation_reason,omitempty"`
	HasMedicalResult   bool            `json:"has_medical_result" gorm:"-"`
}
